libs/sql-parser/src/main/java/io/crate/sql/Identifiers.java

```diff
@@ -80,15 +96,19 @@ public static String quote(String identifier) {
      * quote and escape identifier only if it needs to be quoted to be a valid identifier
      * i.e. when it contain a double-quote, has upper case letters or is a SQL keyword
      */
-    public static String quoteIfNeeded(String identifier) {
-        if (quotesRequired(identifier)) {
```

Comment:
I think we can keep the signature with a single arg, so that it calls a function with the default context and avoid multiple changes passing default context. 

Optional -  we could drop the enum and replace it with a boolean flag

Reply:
Thanks for reviewing @BaurzhanSakhariev .

> I think we can keep the signature with a single arg, so that it calls a function with the default context and avoid multiple changes passing default context.

Then should I rename the method? `quoteIfNeeded(String identifier)` is now misleading I think.

> Optional - we could drop the enum and replace it with a boolean flag

Yes, I will drop it.

Reply:
I think we can keep `quoteIfNeeded(String identifier)` that would use
` if (quotesRequired(false, identifier)) {` - otherwise we still have many unrelated changes just propagating `false`.

And only for Function usages we would introduce `quoteFunctionIfNeeded(String identifier)`  
that would use `if (quotesRequired(true, identifier)) {`